src/sync.d

```diff
@@ -2327,39 +2521,65 @@ final class SyncEngine
 		}
 	}
 	
-	// scan the given directory for differences and new items
+	// scan the given directory for differences and new items - for use with --synchronize
 	void scanForDifferences(const(string) path)
 	{
+		// To improve logging output for this function, what is the 'logical path' we are scanning for file & folder differences?
+		string logPath;
+		if (path == ".") {
+			// get the configured sync_dir
+			logPath = buildNormalizedPath(cfg.getValueString("sync_dir"));
+		} else {
+			// use what was passed in
+			logPath = path;
+		}
+		
 		// Are we configured to use a National Cloud Deployment
 		// Any entry in the DB than is flagged as out-of-sync needs to be cleaned up locally first before we scan the entire DB
 		// Normally, this is done at the end of processing all /delta queries, but National Cloud Deployments (US and DE) do not support /delta as a query
-		if (nationalCloudDeployment) {
+		if ((nationalCloudDeployment) || (syncBusinessFolders)) {
 			// Select items that have a out-of-sync flag set
-			Item[] outOfSyncItems = itemdb.selectOutOfSyncItems();
-			foreach (item; outOfSyncItems) {
-				if (!dryRun) {
-					// clean up idsToDelete
-					idsToDelete.length = 0;
-					assumeSafeAppend(idsToDelete);
-					// flag to delete local file as it now is no longer in sync with OneDrive
-					log.vdebug("Flagging to delete local item as it now is no longer in sync with OneDrive");
-					log.vdebug("item: ", item);
-					idsToDelete ~= [item.driveId, item.id];	
-					// delete items in idsToDelete
-					if (idsToDelete.length > 0) deleteItems();
+			foreach (driveId; uniq(driveIDsArray)) {
```

Comment:
This `uniq(dirveIDsArray)` appears at least two times, and is somehow strange. I would prefer seeing the driveIDsArray uniq-ed during load, which would simplify the code.

Reply:
dirveIDsArray is initialised at the start, then, as each 'drive' is used (default users) + then each drive ID of the shared folder, this is added to the array.

If the user only adds 1 x Shared Folder from the same user, then that drive ID is only in the array once, however, if you add more than 1 shared folder from the same user, that means now you have that drive ID in the array twice.

So when it comes to using those drive ID's (example line 2565), we have:
```
		// For each unique OneDrive driveID we know about
		foreach (driveId; uniq(driveIDsArray)) {
```
If this was changed to:
```
		// For each unique OneDrive driveID we know about
		foreach (driveId; driveIDsArray) {
```
What would happen is that if a drive ID was twice in the array, it would perform a DB scan of all objects for that drive ID X number of times (for the number of times that ID was in the array). If say you have 64K items to process for that drive ID (time intensive) - it would now scan that twice or more redundantly.

The only way I can see to simplify this is at the start of the function doing this:
```
driveIDsArray = uniq(driveIDsArray);
```
But that might have flow on issues ... I would have to check.

Reply:
Also to confuse the situation .. all of these (3384, 3427, 3490, 3694): 
```
				foreach (driveId; driveIDsArray) {
```

Should also be:
```
				foreach (driveId; uniq(driveIDsArray) {
```

For the same reason .. to avoid excessive DB queries

Reply:
https://github.com/abraunegg/onedrive/pull/473/commits/15a845ccba32b9332e38952dbb9896fc144118e9 takes care of this